.cypress/integration/3_panels.spec.ts

```diff
@@ -10,578 +10,624 @@ import {
-      .focus()
-      .type('Renamed Panel', {
-        delay: 200,
+    it('Add ppl filter to panel', () => {
+      createVisualization(
+        PPL_VISUALIZATIONS_NAMES[0],
+        PPL_VISUALIZATIONS[0],
+        PPL_VISUALIZATION_CONFIGS[0]
+      ).as('vis1');
+
+      cy.then(function () {
+        addVisualizationsToPanel(this.thePanel, [this.vis1.id]);
+        moveToThePanel(this.thePanel.id);
+        cy.get('[data-test-subj="breadcrumb"]').click({ force: true });
+        cy.get('input[data-test-subj="operationalPanelSearchBar"]').focus().type(this.thePanel.attributes.title);
+        cy.get('a.euiLink').contains(this.thePanel.attributes.title).click();
       });
-    cy.get('button[data-test-subj="runModalButton"]').click();
-    cy.get('h1[data-test-subj="panelNameHeader"]').contains('Renamed Panel').should('exist');
-  });
 
-  it('Change date filter of the panel', () => {
-    moveToTestPanel();
-    cy.get('.euiButtonEmpty[data-test-subj="superDatePickerToggleQuickMenuButton"]').click({
-      force: true,
+      cy.get('.euiButtonEmpty[data-test-subj="superDatePickerToggleQuickMenuButton"]').click({force: true,});
+      cy.get('[data-test-subj="superDatePickerQuickMenu"')
+        .first()
+        .within(() => {
+          cy.get('input[aria-label="Time value"]').type('2', { force: true });
+          cy.get('select[aria-label="Time unit"]').select('years');
+          cy.get('button').contains('Apply').click();
+        });
+
+      cy.get('[data-test-subj="searchAutocompleteTextArea"]')
+        .trigger('mouseover')
+        .click({force: true})
+        .wait(delay*5)
+        .focus()
+        .type(PPL_FILTER);
+      cy.get('button[data-test-subj="superDatePickerApplyTimeButton"]').click({force: true});
+      cy.get('.euiButton__text').contains('Refresh').trigger('mouseover').click();
+      cy.get('.xtick').should('contain', 'Munich Airport');
+      cy.get('.xtick').contains('Zurich Airport').should('not.exist');
+      cy.get('.xtick').contains('BeatsWest').should('not.exist');
+      cy.get('.xtick').contains('Logstash Airways').should('not.exist');
+      cy.get('.xtick').contains('OpenSearch Dashboards Airlines').should('not.exist');
     });
-    cy.get('button[data-test-subj="superDatePickerCommonlyUsed_This_year"]').click();
-    cy.get('button[data-test-subj="superDatePickerShowDatesButton"]')
-      .contains('This year')
-      .should('exist');
-  });
 
-  it('Add existing visualization #1', () => {
-    cy.get('button[data-test-subj="addVisualizationButton"]').eq(0).click();
+    it('Drag and drop a visualization', () => {
+      createVisualization(
+        PPL_VISUALIZATIONS_NAMES[1],
+        PPL_VISUALIZATIONS[1],
+        PPL_VISUALIZATION_CONFIGS[1]
+      ).as('vis1');
+
+      cy.then(function () {
+        addVisualizationsToPanel(this.thePanel, [this.vis1.id]);
+        moveToThePanel(this.thePanel.id);
+        cy.get('[data-test-subj="breadcrumb"]').click({ force: true });
```

Comment:
When adding a visualization to a panel created from code, forcing a refresh ensures that cypress doesn't execute commands before the visual has appeared on the page.